install/migrations/update_9.5.x_to_10.0.0/databases.php

```diff
@@ -168,11 +168,17 @@
       ) ENGINE = InnoDB ROW_FORMAT = DYNAMIC DEFAULT CHARSET = {$default_charset} COLLATE = {$default_collation};";
     $DB->queryOrDie($query, "10.0 add table glpi_databases");
 }
-$migration->addField('glpi_states', 'is_visible_database', 'bool', [
-    'value' => 1,
-    'after' => 'is_visible_appliance'
-]);
-$migration->addKey('glpi_states', 'is_visible_database');
+
+if ($DB->fieldExists('glpi_states', 'is_visible_database')) {
+    // Dev migration
+    $migration->changeField('glpi_states', 'is_visible_database', 'is_visible_databaseinstance', 'bool', ['value' => 1]);
+} else if (!$DB->fieldExists('glpi_states', 'is_visible_databaseinstance')) {
+    $migration->addField('glpi_states', 'is_visible_databaseinstance', 'bool', [
+        'value' => 1,
+        'after' => 'is_visible_appliance'
+    ]);
+    $migration->addKey('glpi_states', 'is_visible_databaseinstance');
+}
```

Comment:
To prevent any failure on a potential future action on this index, we have to update its name. This could be achieved by dropping the old one and then create the new one.
```suggestion
if ($DB->fieldExists('glpi_states', 'is_visible_database')) {
    // Dev migration
    $migration->changeField('glpi_states', 'is_visible_database', 'is_visible_databaseinstance', 'bool', ['value' => 1]);
    $migration->dropKey('glpi_states', 'is_visible_database');
} else if (!$DB->fieldExists('glpi_states', 'is_visible_databaseinstance')) {
    $migration->addField('glpi_states', 'is_visible_databaseinstance', 'bool', [
        'value' => 1,
        'after' => 'is_visible_appliance'
    ]);
}
$migration->addKey('glpi_states', 'is_visible_databaseinstance');
```